models/select-all-articles.model.js

```diff
@@ -12,16 +13,26 @@ exports.selectAllArticles = async () => {
             COUNT(comments.article_id) AS comment_count 
         FROM articles 
         LEFT JOIN comments 
-        ON articles.article_id = comments.article_id
-        GROUP BY 
+        ON articles.article_id = comments.article_id`;
+
+    if(topic) {
+        queryString += ` WHERE topic = $1`;
+        queryArray.push(topic);
+    }
+
+    queryString += ` GROUP BY 
             articles.article_id, 
             articles.title, 
             articles.topic, 
             articles.author, 
             articles.created_at, 
             articles.votes, 
             articles.article_img_url
-        ORDER BY articles.created_at DESC;`
-    );
-    return articles.rows;
+        ORDER BY articles.created_at DESC;`;
+
+    const result = await db.query(queryString, queryArray);
+    if (result.rows.length === 0) {
+        return Promise.reject({statusCode: 404, message: `No articles for topic: ${topic}`});
+    }
```

Comment:
This may cause issues when we add our tests above as an empty array here isn't strictly an error.